Update internal-tools.md (#23716)
department-of-veterans-affairs/va.gov-team: platform/engineering/internal-tools.md

The `~/.ssh/config` file on your local system contains configuration to access the SOCKS proxy from both [inside](https://github.com/department-of-veterans-affairs/devops/blob/master/ssh/config#L34) and [outside](https://github.com/department-of-veterans-affairs/devops/blob/master/ssh/config#L28) the VA network.

> 

Reviewer: i think you want this section to look something like this:

```markdown

> The first time you connect to the proxy, SSH will prompt to ask if you are sure you want to connect to a new host. You will have to enter "yes" at the prompt for the first / initial connection.

- From inside VA network:

   `ssh socks-va -D 2001 -N`

- From outside VA network:

   `ssh socks -D 2001 -N`

> Once the script is running, you can put it in the background by typing `control-z`. In the future you can add an ampersand after the command to keep it running in the background.

- From inside VA network:

   `ssh socks-va -D 2001 -N &`

- From outside VA network:

   `ssh socks -D 2001 -N &`

```

The `~/.ssh/config` file on your local system contains configuration to access the SOCKS proxy from both [inside](https://github.com/department-of-veterans-affairs/devops/blob/master/ssh/config#L34) and [outside](https://github.com/department-of-veterans-affairs/devops/blob/master/ssh/config#L28) the VA network.

Run the following command:

#### from _inside_ the VA network:

   `ssh socks-va -D 2001 -N`

#### from _outside_ the VA network:
   
   `ssh socks -D 2001 -N`
   
>**Note:** The first time you connect to the jumpbox, SSH will prompt to ask if you are sure you want to connect to a new host. You will have to enter "yes" at the prompt for the first / initial connection.

Once the script is running, you can put it in the background by typing `control-z`. In the future you can add an ampersand after the command to keep it running in the background:

#### from _inside_ the VA network:

   `ssh socks-va -D 2001 -N &`

> #### from _outside_ the VA network:

Reviewer: ```suggestion
#### From _outside_ the VA network:
```

You should get output that includes `HTTP/1.1 302 FOUND`. If not, check that the SOCKS proxy server is running. You can run `$ nc -z 127.0.0.1 2001` as a first step.

### Connecting your network to the proxy
There are two alternatives to connecting your PC's network to the proxy. (Mac OS users only):

> 1. Use the `socks.sh` script: a script will set up OSX with a system-level proxy

Reviewer: what script is this?

> 1. Install an app called [CoreTunnel](https://coretunnel.app/) that does the same thing as the `socks.sh` script in a more user-friendly way. See this [video](https://www.youtube.com/watch?v=fSuN9LhkB5o) for setup instructions.

Reviewer: this is a really helpful video! is it possible to also write out the steps for reference?

You should get output that includes `HTTP/1.1 302 FOUND`. If not, check that the SOCKS proxy server is running. You can run `$ nc -z 127.0.0.1 2001` as a first step.

### Connecting your network to the proxy

> There are two alternatives to connecting your PC's network to the proxy. (Mac OS users only):

Reviewer: ```suggestion
#### Mac OSX
There are two alternatives to connecting your PC's network to the proxy:
```

1. Use the [socks.sh script](https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/scripts/socks): this will set up OSX with a system-level proxy

> 1. Install an app called [CoreTunnel](https://coretunnel.app/) that does the same thing as the `socks.sh` script in a more user-friendly way. See this [video](https://www.youtube.com/watch?v=fSuN9LhkB5o) for setup instructions. 
> 
> Here is a summary:
> 
>     1. Download CoreTunnel from the [App Store](https://apps.apple.com/us/app/core-tunnel/id1354318707). Launch CoreTunnel.
>     2. Add Private Key (`~/.ssh/id_rsa_vagov`) in Preferences (if not found automatically)
>     3. Make sure Enable Core Helper is checked under Advanced (this may require a separate download)
>     4. Add CoreTunnel to your Login Items in System Preferences / Users so it launches automatically when you log into your Mac
>     5. Click + to add a connection: Name= VA, host=socks, Forwarding=Dynamic, port=2001
>     6. Click "automatically connect on startup"
>     7. Configure your WiFi and Ethernet Network connections in system preferences / Network / Advanced. Select Automaic Proxy Configuration and enter the following URL for the Proxy Configuration File: https://raw.githubusercontent.com/department-of-veterans-affairs/va.gov-team/master/scripts/socks/proxy.pac

Reviewer: ```suggestion
1. Install an app called [CoreTunnel](https://coretunnel.app/) that does the same thing as the `socks.sh` script in a more user-friendly way. See this [video](https://www.youtube.com/watch?v=fSuN9LhkB5o) for setup instructions.
Summary:
    1. Download CoreTunnel from the [App Store](https://apps.apple.com/us/app/core-tunnel/id1354318707). Launch CoreTunnel.
    2. Add Private Key (`~/.ssh/id_rsa_vagov`) in Preferences (if not found automatically)
    3. Make sure Enable Core Helper is checked under Advanced (this may require a separate download)
    4. Add CoreTunnel to your Login Items in System Preferences / Users so it launches automatically when you log into your Mac
    5. Click + to add a connection: Name= VA, host=socks, Forwarding=Dynamic, port=2001
    6. Click "automatically connect on startup"
    7. Configure your WiFi and Ethernet Network connections in system preferences / Network / Advanced. Select Automaic Proxy Configuration and enter the following URL for the Proxy Configuration File: https://raw.githubusercontent.com/department-of-veterans-affairs/va.gov-team/master/scripts/socks/proxy.pac
```

trying to nest this into parent list, not sure if i did it quite right...
